### PR TITLE
added UPDATE support

### PR DESCRIPTION
**Pull request by juha-h**
Requires also commit of corresponding pull request to re.

### CODE REVIEW

File: src/call.c [hunk 1]
```diff
@@ -1067,7 +1067,12 @@ static int sipsess_offer_handler(struct mbuf **descp,
 
 	MAGIC_CHECK(call);
 
-	info("call: got re-INVITE%s\n", got_offer ? " (SDP Offer)" : "");
+	if (!pl_strcmp(&msg->met, "INVITE")) {
+		info("call: got re-INVITE%s\n",
+		     got_offer ? " (SDP Offer)" : "");
+	} else {
```
Comment (alfredh) [inline src/call.c]: perhaps we should also check that the method is UPDATE here ?

**Comment by alfredh** [comment]
1 ccheck warning must be fixed:

```
./src/call.c:1073: else: ending if bracket should be on previous line
```

File: src/ua.c [hunk 2]
```diff
@@ -1843,7 +1843,7 @@ struct list *uag_list(void)
 const char *uag_allowed_methods(void)
 {
 	return "INVITE,ACK,BYE,CANCEL,OPTIONS,REFER,"
-		"NOTIFY,SUBSCRIBE,INFO,MESSAGE";
+		"NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE";
```
Comment (alfredh) [inline src/ua.c]: this assumes that the version of libre used supports UPDATE.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> > @@ -1067,7 +1067,12 @@ static int sipsess_offer_handler(struct mbuf **descp,
>  
>  	MAGIC_CHECK(call);
>  
> -	info("call: got re-INVITE%s\n", got_offer ? " (SDP Offer)" : "");
> +	if (!pl_strcmp(&msg->met, "INVITE")) {
> +		info("call: got re-INVITE%s\n",
> +		     got_offer ? " (SDP Offer)" : "");
> +	} else {
> 
> perhaps we should also check that the method is UPDATE here ?

How about if it is neither?  Currently there is no check at all.

-- Juha

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> ./src/call.c:1073: else: ending if bracket should be on previous line

Will fix.  How do you get those warnings?

-- Juha

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> > @@ -1843,7 +1843,7 @@ struct list *uag_list(void)
>  const char *uag_allowed_methods(void)
>  {
>  	return "INVITE,ACK,BYE,CANCEL,OPTIONS,REFER,"
> -		"NOTIFY,SUBSCRIBE,INFO,MESSAGE";
> +		"NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE";
> 
> this assumes that the version of libre used supports UPDATE.

Yes, I mentioned it in the comment, i.e., UPDATE support in baresip
requires committing the corresponding re pull request.

-- Juha

**Comment by alfredh** [comment]
regarding ccheck:

https://github.com/alfredh/pytools/blob/master/ccheck.py

or look at travis CI build log.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> > @@ -1067,7 +1067,12 @@ static int sipsess_offer_handler(struct mbuf **descp,
>  
>  	MAGIC_CHECK(call);
>  
> -	info("call: got re-INVITE%s\n", got_offer ? " (SDP Offer)" : "");
> +	if (!pl_strcmp(&msg->met, "INVITE")) {
> +		info("call: got re-INVITE%s\n",
> +		     got_offer ? " (SDP Offer)" : "");
> +	} else {
> 
> perhaps we should also check that the method is UPDATE here ?

This has now been fixed.  Info about any method will get printed.

-- Juha

**Comment by alfredh** [comment]
there is a small ccheck error that must be fixed first:

./src/call.c:1073: else: ending if bracket should be on previous line

**Comment by juha-h** [comment]
I'll close this one and create a new one to replace it.

**Pull request closed without merging**